sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/percentiles.scala

```diff
@@ -49,6 +50,9 @@ abstract class PercentileBase
   @transient
   private lazy val returnPercentileArray = percentageExpression.dataType.isInstanceOf[ArrayType]
 
+  private val legacyDiscCalculation: Boolean =
+    SQLConf.get.getConf(SQLConf.LEGACY_PERCENTILE_DISC_CALCULATION)
```

Comment:
This might not work well, see:
- the reasons for moving val to constructor params in https://github.com/apache/spark/pull/27658
- https://github.com/apache/spark/pull/27628#discussion_r381246285

Reply:
Ah ok, thanks. I will move this to class param then.

Reply:
I've refactored the PR a bit so that only `PercentileDisc` requires the new legacy flag.